commons/src/main/java/eu/itesla_project/commons/config/PlatformConfig.java

```diff
@@ -91,9 +83,84 @@ public static synchronized CacheManager defaultCacheManager() {
         return defaultCacheManager;
     }
 
+    protected PlatformConfig(FileSystem fileSystem) {
+        this(fileSystem, getDefaultConfigDir(fileSystem), getDefaultCacheDir(fileSystem));
+    }
+
+    protected PlatformConfig(FileSystem fileSystem, Path configDir, Path cacheDir) {
+        this.fileSystem = Objects.requireNonNull(fileSystem);
+        this.configDir = createConfigDir(configDir);
+        this.cacheDir = createCacheDir(cacheDir);
+    }
+
+    public FileSystem getFileSystem() {
+        return fileSystem;
+    }
+
+    public Path getConfigDir() {
+        return configDir;
+    }
+
+    public Path getCacheDir() {
+        return cacheDir;
+    }
+
     public abstract boolean moduleExists(String name);
 
     public abstract ModuleConfig getModuleConfig(String name);
 
     public abstract ModuleConfig getModuleConfigIfExists(String name);
+
+    static Path getDefaultConfigDir(FileSystem fileSystem) {
+        Objects.requireNonNull(fileSystem);
+
+        Path configDir;
+
+        String iteslaConfigDir = System.getProperty("itesla.config.dir");
+        if (iteslaConfigDir != null) {
+            configDir = fileSystem.getPath(iteslaConfigDir);
+        } else {
+            configDir = fileSystem.getPath(System.getProperty("user.home"), ".itesla");
+        }
+
+        return configDir;
+    }
+
+    private static Path createConfigDir(Path configDir) {
```

Comment:
createConfigDir and createCacheDir are same function. This is just Files.createDirectory with unchecked exception.